lib/Parser/Scan.cpp

```diff
@@ -2087,36 +2092,14 @@ tokens Scanner<EncodingPolicy>::ScanCore(bool identifyKwds)
             case '-':
                 p++;
                 token = tkDec;
-                if (m_fHtmlComments)
+                if (!m_fIsModuleCode)
                 {
                     int i = 0;
                     while ('-' == PeekFirst(p + i, last)) //Have already seen --, skip any further - characters
                         i++;
-                    if ('>' == PeekFirst(p + i++, last)) //This means we've got a --------------------------->.
+                    if ('>' == PeekFirst(p + i++, last) && m_fHadEol) //This means we've got a --------------------------->.
```

Comment:
`i++` is subtle.  `i` is no longer used afterwards so its moot but I'd be happier reading this code if this was `p + i + 1` instead.